principal/apis/eventstream/eventstream.go

```diff
@@ -295,10 +247,14 @@ func (s *Server) sendFunc(c *client, subs eventstreamapi.EventStream_SubscribeSe
 	// Get() is blocking until there is at least one item in the
 	// queue.
 	logCtx.Tracef("Waiting to grab an item from the queue")
-	ev, shutdown := q.Get()
+	ev, shutdown := queue.GetWithContext(q, c.ctx)
 	if shutdown {
```

Comment:
Get() is a blocking call, and it suspends until a new item is added to the queue or if the queue shuts down. Since we don't remove the queue on disconnect, Get will never return. So, I've added a wrapper around Get() that checks if the context is cancelled. So, on disconnect, the client context is cancelled, and the Get should no longer block.